src/main/cpp/src/config/json_config.cc

```diff
@@ -358,46 +359,80 @@ void JSONConfigBase::read_from_file(const std::string& filename, const int rank)
                     "Cannot use both \"query_row_ranges\" and \"row_partitions\" simultaneously");
+    VERIFY_OR_THROW((!m_json.HasMember("query_sample_names_lists") || !m_json.HasMember("row_partitions")) &&
+                    "Cannot use both \"query_sample_names_lists\" and \"row_partitions\" simultaneously");
     //Query rows
     //Example:  [ [ [0,5], 45 ], [ 76, 87 ] ]
     //This means that rank 0 will query rows: [0-5] and [45-45] and rank 1 will have
     //2 intervals [76-76] and [87-87]
     //But you could have a single innermost list - with this option all ranks will query the same list
-    if (m_json.HasMember("query_row_ranges")) {
-      const rapidjson::Value& q1 = m_json["query_row_ranges"];
-      VERIFY_OR_THROW(q1.IsArray());
-      if (q1.Size() == 1)
-        m_single_query_row_ranges_vector = true;
-      m_row_ranges.resize(q1.Size());
-      for (rapidjson::SizeType i=0; i<q1.Size(); ++i) {
-        auto& curr_q1_entry = q1[i];
-        VERIFY_OR_THROW(curr_q1_entry.IsArray() || curr_q1_entry.IsObject());
-        //JSON produced by Protobuf - { "range_list": [ { "low":<>, "high":<> } ] }
-        if (curr_q1_entry.IsObject())
-          VERIFY_OR_THROW(curr_q1_entry.MemberCount() == 1u && curr_q1_entry.HasMember("range_list"));
-        const rapidjson::Value& q2 = curr_q1_entry.IsArray() ? q1[i]
-                                     : curr_q1_entry["range_list"];
-        VERIFY_OR_THROW(q2.IsArray());
-        m_row_ranges[i].resize(q2.Size());
-        for (rapidjson::SizeType j=0; j<q2.Size(); ++j) {
-          const rapidjson::Value& q3 = q2[j];
-          //q3 is list of 2 elements to represent query row interval
-          if (q3.IsArray()) {
-            VERIFY_OR_THROW(q3.Size() == 2);
-            VERIFY_OR_THROW(q3[0u].IsInt64());
-            VERIFY_OR_THROW(q3[1u].IsInt64());
-            m_row_ranges[i][j].first = q3[0u].GetInt64();
-            m_row_ranges[i][j].second = q3[1u].GetInt64();
-          } else {
-            if (q3.IsInt64()) { //single position
-              m_row_ranges[i][j].first = q3.GetInt64();
-              m_row_ranges[i][j].second = q3.GetInt64();
-            } else if (q3.IsObject()) //Must be PB generated JSON object { "low": <>, "high": <> }
-              VERIFY_OR_THROW(extract_interval_from_PB_struct_or_return_false(q3, &m_vid_mapper, m_row_ranges[i][j]));
-          }
-          if (m_row_ranges[i][j].first > m_row_ranges[i][j].second)
-            std::swap<int64_t>(m_row_ranges[i][j].first, m_row_ranges[i][j].second);
-        }
+    if (m_json.HasMember("query_row_ranges") || m_json.HasMember("query_sample_names_lists")) {
+      if(m_json.HasMember("query_row_ranges") && m_json.HasMember("query_sample_names_lists"))
+	VERIFY_OR_THROW(m_json["query_row_ranges"].Size() == m_json["query_sample_names_lists"].Size());
+      for(const auto json_name : { "query_row_ranges", "query_sample_names_lists" })
+      {
```

Comment:
Extremely minor quibble. Would it be possible to use the Google style of parentheses - https://google.github.io/styleguide/cppguide.html?

Reply:
Done